File: web/src/types/patient.ts
import type { Specimen } from "./specimen";

export interface Patient {
  id: number;
  first_name: string;
  last_name: string;
  birthdate: string;
  sex: string;
  phone_number: string;
  location: string;
  address: string;
  created_at: string;
  updated_at: string;
  specimen_list: Specimen[] | null;
}

File: integration-service/internal/models/patient.go
package models

import (
	"database/sql"
	"time"
)

type PatientSex string

const (
	PatientSexMale    PatientSex = "M"
	PatientSexFemale  PatientSex = "F"
	PatientSexUnknown PatientSex = "U"
)

type Patient struct {
	ID                  int64
	FirstName           string
	LastName            string
	Birthdate           time.Time
	Sex                 PatientSex
	PhoneNumber         string
	Location            string
	Address             string
	CreatedAt           time.Time
	UpdatedAt           time.Time
	SimrsPid            sql.NullString
	MedicalRecordNumber string
}
